src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,341 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlTools.ServiceLayer.LanguageExtensibility.Contracts;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageExtensibility
+{
+
+    public enum ModifyType
+    {
+        Create,
+        Alter
+    }
+
+    public enum ContentModifyType
+    {
+        Modify,
+        Add,
+        Remove
+    }
+
+    public class ExternalLanguageOperations
+    {
+        private static string StatusScript = $@"
+SELECT is_installed
+FROM sys.dm_db_external_language_stats s, sys.external_languages l
+WHERE s.external_language_id = l.external_language_id AND language = @{LanguageNameParamName}";
+
+        private const string GetAllScript = @"
+SELECT l.external_language_id, language, l.create_date, dp.name, content, file_name, platform_desc, parameters, environment_variables
+FROM sys.external_languages l 
+JOIN sys.external_language_files lf on l.external_language_id = lf.external_language_id
+JOIN sys.database_principals dp on l.principal_id = dp.principal_id
+ORDER BY l.external_language_id, platform";
+
+        private static string GetLanguageScript = $@"
+SELECT l.external_language_id, language, l.create_date, dp.name, content, file_name, platform_desc, parameters, environment_variables
+FROM  sys.external_languages l 
+JOIN sys.external_language_files lf on l.external_language_id = lf.external_language_id
+JOIN sys.database_principals dp on l.principal_id = dp.principal_id
+WHERE language=@{LanguageNameParamName}
+ORDER BY platform";
+
+        public const string CreateScript = "CREATE EXTERNAL LANGUAGE";
+        public const string DropScript = "DROP EXTERNAL LANGUAGE";
+        public const string AlterScript = "ALTER EXTERNAL LANGUAGE";
+        public const string SetContentScript = "SET";
+        public const string AddContentScript = "ADD";
+        public const string RemoveContentScript = "REMOVE";
+        private const string ContentParamName = "CONTENT";
+        private const string FileNameParamName = "FILE_NAME";
+        private const string PlatformParamName = "PLATFORM";
+        private const string EnvVariablesParamName = "ENVIRONMENT_VARIABLES";
+        private const string ParametersParamName = "PARAMETERS";
+        private const string LanguageNameParamName = "LANGUAGE";
+
+        private string GetDropScript(string languageName)
+        {
+            return $@"{DropScript} [{languageName}]";
```

Comment:
You've verified that bracket-quoting the language name is supported? And if it is you should also be escaping the ]'s for the same reason as the single-quotes. (Even if currently languages are hard-coded with values that don't include a ] it's still better to be safe and do that here now in case that changes later and we allow users to enter in their own language names)